src/reactapp/index.html

```diff
@@ -62,6 +51,8 @@
         </div>
     </main>
 </div>
+<script type="module" src="./src/index.jsx"></script>
```

Comment:
How will this work with browsers, who do not support jsx files? Shouldn't this point to the generated js files?

Reply:
This file is there only for development, for production it compiles a .js file which is referenced to in the layout file.

'index.html' in Vite is being treated as source code and part of the module graph. It will resolve the .jsx file. It doesn't create a bundle when developing which makes it super quick.